tests/lib/tools/tests.session

```diff
@@ -32,10 +32,6 @@ main() {
 				pid_file="$2"
 				shift 2
 				;;
-			--)
-				shift
-				break
-				;;
```

Comment:
why this change? seems a bit unrelated, also @sergiocazzolato should review

Reply:
The existing help text didn't actually work:

```
show_help() {
    echo "usage: tests.session exec [-u USER] [-p PID_FILE] [--] <CMD>"
    echo "       tests.session prepare | restore [-u USER | -u USER1,USER2,...]"
    echo "       tests.session kill-leaked"
    echo "       tests.session dump"
    echo "       tests.session has-system-systemd-and-dbus"
    echo "       tests.session has-session-systemd-and-dbus"
}
```

Here's the switch for the exec keyword:
```
            exec) 
                action="exec"
                shift
                # remaining arguments are the command to execute
                break
                ;;
```

So if you try to do `tests.session exec -u test -- echo foo`, it'll treat `-u`, `test`, `--` `echo` `foo` as the command to execute, which is wrong.

So I saw two options:
1. change the behavior of `exec` so it actually expects `--` and doesn't `break` immediately
2. remove the `--` from the `exec` help text

I checked every usage of `tests.session` in spread tests, and all of them put `-u USER -p PID_FILE` before the `exec`. So I did the latter with regards to `exec`. And `--` isn't used by any other command, so I thought it best to remove it.

Reply:
Agree, the help needs an update, after the exec command it breaks, so what it expected is the cmd to execute.

Reply:
other way to do this is removing the break in exec, this will force to use -- to delimit the command to execute.